src/components/AttachmentCarousel/index.js

```diff
@@ -1,398 +1,318 @@
+        cancelAutoHideArrow();
+        autoHideArrowTimeout.current = setTimeout(() => {
+            toggleArrowsVisibility(false);
+        }, CONST.ARROW_HIDE_DELAY);
+    }, [cancelAutoHideArrow, toggleArrowsVisibility]);
+
+    useEffect(() => {
+        if (shouldShowArrow) {
+            autoHideArrow();
+        } else {
+            cancelAutoHideArrow();
         }
-        this.setState({isZoomed});
-    }
+    }, [shouldShowArrow, autoHideArrow, cancelAutoHideArrow]);
 
     /**
-     * Constructs the initial component state from report actions
-     * @returns {{page: Number, attachments: Array, shouldShowArrow: Boolean, containerWidth: Number, isZoomed: Boolean}}
+     * Updates zoomed state to enable/disable panning the PDF
+     * @param {Number} scale current PDF scale
      */
-    createInitialState() {
-        const actions = [ReportActionsUtils.getParentReportAction(this.props.report), ...ReportActionsUtils.getSortedReportActions(_.values(this.props.reportActions))];
-        const attachments = [];
-
-        const htmlParser = new HtmlParser({
-            onopentag: (name, attribs) => {
-                if (name !== 'img' || !attribs.src) {
-                    return;
-                }
-
-                const expensifySource = attribs[CONST.ATTACHMENT_SOURCE_ATTRIBUTE];
-
-                // By iterating actions in chronological order and prepending each attachment
-                // we ensure correct order of attachments even across actions with multiple attachments.
-                attachments.unshift({
-                    source: tryResolveUrlFromApiRoot(expensifySource || attribs.src),
-                    isAuthTokenRequired: Boolean(expensifySource),
-                    file: {name: attribs[CONST.ATTACHMENT_ORIGINAL_FILENAME_ATTRIBUTE]},
-                });
-            },
-        });
-
-        _.forEach(actions, (action, key) => {
-            if (!ReportActionsUtils.shouldReportActionBeVisible(action, key)) {
+    const updateZoomState = useCallback(
+        (scale) => {
+            const newIsZoomed = scale > 1;
+            if (newIsZoomed === isZoomed) {
                 return;
             }
-            htmlParser.write(_.get(action, ['message', 0, 'html']));
-        });
-        htmlParser.end();
-
-        // Inverting the list for touchscreen devices that can swipe or have an animation when scrolling
-        // promotes the natural feeling of swiping left/right to go to the next/previous image
-        // We don't want to invert the list for desktop/web because this interferes with mouse
-        // wheel or trackpad scrolling (in cases like document preview where you can scroll vertically)
-        if (this.canUseTouchScreen) {
-            attachments.reverse();
-        }
-
-        const page = _.findIndex(attachments, (a) => a.source === this.props.source);
-        if (page === -1) {
-            throw new Error('Attachment not found');
-        }
```

Comment:
The original code here would have lead to the "Ooops" screen.

During the code refactoring, the error was replaced with the immediate dismissal of the attachment modal. I believe presenting the App error screen offered a more user-friendly experience.

Reply:
Thanks for taking the time and pointing me to the correct PR @fedirjh